app/routes.py

```diff
@@ -1,2 +1,191 @@
-from flask import Blueprint
+from flask import Blueprint, jsonify, request, make_response, abort
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import date
+import requests
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+# Blueprints instantiated 
+tasks_bp = Blueprint("tasks_bp",__name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+# helper function to query one task and return 404 if falsey
+def query_one_task(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        abort(404)
+    else:
+        return task
+
+# helper function to query one goal and return 404 if falsey
+def query_one_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        abort(404)
+    else:
+        return goal
+
+# TASK ROUTES
+# add new task to database
+@tasks_bp.route('', methods=["POST"])
+def new_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"],
+        completed_at = request_body["completed_at"]
+    )
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+# gets all tasks
+@tasks_bp.route('', methods=["GET"])
+def get_all_tasks():
+    response_tasks = []
+    tasks = Task.query.all()
+
+    # sort ascending or descending if query param sort present
+    if request.args.get("sort") == "asc":
+        tasks = Task.query.order_by(Task.title)
+    elif request.args.get("sort") == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+
+    for task in tasks:
+        response_tasks.append(task.to_dict())
```

Comment:
This works well and is very readable, but if you'd like to incorporate list comprehensions into your code, a loop that builds a list like this is a great place to start:
```suggestion
    response_tasks = [task.to_dict() for task in tasks]
```